obs-server/check_new_uploads

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python3
+
+import imaplib
+import email
+import subprocess
+import time
+
+TRIGGER_BIN_PATH = '/usr/local/bin/trigger_clang_build'
+
+# Hours to wait before triggering a build;
+# this allows propagation time so the package is available in source mirrors.
+COOLDOWN = 6
+
+# Mail server IMAP information and credentials;
+# this must be an email subscribed to the debian-devel-changes mailing list.
+IMAP_SERVER = "{{ salt['pillar.get']('email-credentials:lookup:hostname') }}"
+USERNAME = "{{ salt['pillar.get']('email-credentials:lookup:user') }}"
+PASSWORD = "{{ salt['pillar.get']('email-credentials:lookup:pass') }}"
+
+
+def parse_message(msg):
+    lines = msg.split('\n')
+    mapping = {
+        'Source': 'name',
+        'Version': 'version',
+        'Architecture': 'arch',
+        'Distribution': 'dist',
+        'Maintainer': 'maintainer'
+    }
+    pkg = {}
+    for line in lines:
+        if len(mapping) == 0:
+            break
+        for field, target in mapping.items():
+            if line.startswith('%s: ' % field):
+                val = line[len(field) + 2:]
+                pkg[target] = val.strip()
+                del mapping[field]
+                break
+    return pkg
+
+
+def trigger_build(pkg):
+    subprocess.run([TRIGGER_BIN_PATH, pkg['name']])
+
+
+subprocess.run(['apt', 'update'])
```

Comment:
why do you have to do it?

Reply:
As of now, we are checking emails every 2 hours. Don't we need to update the cache to fetch packages that were just uploaded?

Reply:
This requires to be root. I would prefer you use sudo for that